src/test/java/org/assertj/core/internal/paths/Paths_assertHasSize_Test.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2021 the original author or authors.
+ */
+package org.assertj.core.internal.paths;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.assertj.core.api.BDDAssertions.then;
+import static org.assertj.core.error.ShouldExist.shouldExist;
+import static org.assertj.core.error.ShouldHaveSize.shouldHaveSize;
+
+import static org.assertj.core.util.AssertionsUtil.expectAssertionError;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+import static org.mockito.BDDMockito.given;
+
+import org.assertj.core.api.AssertionInfo;
+import org.assertj.core.internal.Paths;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Path;
+
+/**
+ * Tests for <code>{@link Paths#assertHasSize(AssertionInfo, Path, long)} </code>
+ */
+@DisplayName("Paths assertHasSize")
+class Paths_assertHasSize_Test extends MockPathsBaseTest {
```

Comment:
We decided to slowly decommission the `MockPathsBaseTest` and the related testing strategy, as relying on `java.nio.Files` methods with JUnit 5 `@TempDir` should be enough. Being this a new feature, I'd try to avoid `MockPathsBaseTest` usage.